apps/demos/Demos/Chat/Customization/React/App.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useState, useCallback } from 'react';
+import Chat, { ChatTypes } from 'devextreme-react/chat';
+import { MessageEnteredEvent } from 'devextreme/ui/chat';
+import SelectBox, { SelectBoxTypes } from 'devextreme-react/select-box';
+import CheckBox, { CheckBoxTypes } from 'devextreme-react/check-box';
+
+import { 
+    currentUser,
+    messages as initialMessages,
+    dayHeaderFormat as headerFormats,
+    messageTimestampFormat  as msgTimestamp
+  } from './data.ts';
+
+export default function App() {
+  const [messages, setMessages] = useState<ChatTypes.Message[]>(initialMessages);
+  const [showAvatar, setShowAvatar] = useState(true);
+  const [showUsername, setShowUsername] = useState(true);
+  const [showDayHeaders, setDayHeaders] = useState(true);
+  const [dayHeaderFormat, setDayHeaderFormat] = useState(headerFormats[0]);
+  const [showMsgTime, setMsgTime] = useState(true);
+  const [messageTimestampFormat, setMsgTimeFormat] = useState(msgTimestamp[0]);
+  const [isDisabled, setDisabled] = useState(false);
+
+  function onMessageEntered({ message }: MessageEnteredEvent) {
+    setMessages(prevMessages => [...prevMessages, message]);
+  }
+
+  const dayHeaderFormatChanged = useCallback((e: SelectBoxTypes.ValueChangedEvent) => {
+    setDayHeaderFormat(e.value);
+  }, [setDayHeaderFormat]);
+
+  const messageTimestampFormatChanged = useCallback((e: SelectBoxTypes.ValueChangedEvent) => {
+    setMsgTimeFormat(e.value);
+  }, [setMsgTimeFormat]);
+
+  const showAvatarChanged = useCallback((e: CheckBoxTypes.ValueChangedEvent) => {
+    setShowAvatar(e.value);
+  }, [setShowAvatar]);
+
+  const showUsernameChanged = useCallback((e: CheckBoxTypes.ValueChangedEvent) => {
+    setShowUsername(e.value);
+  }, [setShowUsername]);
+
+  const showDayHeaderChanged = useCallback((e: CheckBoxTypes.ValueChangedEvent) => {
+    setDayHeaders(e.value);
+  }, [setDayHeaders]);
+
+  const showMsgTimeChanged = useCallback((e: CheckBoxTypes.ValueChangedEvent) => {
+    setMsgTime(e.value);
```

Comment:
Let's avoid abbreviations, e.g., use 'Message' instead of 'Msg

Reply:
[a40d776](https://github.com/DevExpress/DevExtreme/pull/28482/commits/a40d77698d64bcb42c0340010e88cdccd0628c50)